test/commands/scanner/rule/list.test.ts

```diff
@@ -55,13 +61,15 @@ describe('scanner:rule:list', () => {
 		describe('Test Case: Filtering by category only', () => {
 			setupCommandTest
 				.command(['scanner:rule:list', '--category', 'Best Practices', '--json'])
-				.it('Filtering by one category returns only the rules in that category', ctx => {
+				.it('Filtering by one category returns only the rules in that category for enabled engines', async ctx => {
 					// Rather than painstakingly checking everything about all the rules, we'll just make sure that the number of rules
 					// returned is the same as the number of rules in the target category, and that every rule returned is actually
 					// a member of that category.
 					// The first step is to identify how many satisfactory rules are in the catalog.
 					const catalog = getCatalogJson();
-					const targetRuleCount = catalog.rules.filter(rule => rule.categories.includes('Best Practices')).length;
+					const enabledEngines = (await Controller.getEnabledEngines()).map(e => e.getName());
+					const filterFunction: (Rule) => boolean = (r: Rule) => r.categories.includes('Best Practices') && enabledEngines.includes(r.engine);
+					const targetRuleCount = catalog.rules.filter(filterFunction).length;
```

Comment:
Programmatic tests are good, but they can themselves have bugs that cause false passes. Can you add an assertion that targetRuleCount > 0 to ensure that the test is testing what we expect. Same request where this is used below.

Reply:
Will do.